PyPoE/poe/constants.py

```diff
@@ -585,10 +585,10 @@ class RARITY(Enum, metaclass=IntEnumMetaOverride):
         the colour attribute represents the textual representation of the
         associated colour
     """
-    NORMAL = (1, 'Normal', 'normal', 'white')
-    MAGIC = (2, 'Magic', 'magic', 'blue')
-    RARE = (3, 'Rare', 'rare', 'yellow')
-    UNIQUE = (4, 'Unique', 'unique', 'brown')
+    NORMAL = (0, 'Normal', 'normal', 'white')
+    MAGIC = (1, 'Magic', 'magic', 'blue')
+    RARE = (2, 'Rare', 'rare', 'yellow')
+    UNIQUE = (3, 'Unique', 'unique', 'brown')
```

Comment:
This is more of a question to you rather than a suggestion:

Would it make sense to remove the uppercased entries out of here? It feels like this is unnecessary.
For instance if you are comparing incoming text which CAN be any case with this class, you can easily do:

```py
...
NORMAL = (0, 'normal', 'white')
...
```

then

`#1`
```py
some_text = "This item is of Normal rarity"
RARITY.NORMAL[1] in some_text.lower()
```
`#2`
```py
some_row = [0, "Normal", "White"]
for col in some_row:
  if isinstance(col, int):
    # do something
  if col.lower() == RARITY.NORMAL[1]:
    ..
```

I guess what I am asking is - does it make it cleaner if we remove what seems to be a redundant capitalised entry and homogenise incoming data for this by using `.lower()` on it for any comparisons with this class?

Reply:
I think it's better to have one string for the rarity in the `RARITY` enum values, and use `.lower()` and `.title()` as needed.

I went ahead and cleaned up the enum.
Figured out how to add typehints for the contents of the enum values as well.